main.py

```diff
@@ -456,37 +500,13 @@ def main(cl_arguments):
 
     if args.do_target_task_training:
         # Train on target tasks
-        task_names_to_avoid_loading = setup_target_task_training(args, target_tasks, model, strict)
-        if args.transfer_paradigm == "frozen":
-            # might be empty if elmo = 0. scalar_mix_0 should always be
-            # pretrain scalars
-            elmo_scalars = [
-                (n, p)
-                for n, p in model.named_parameters()
-                if "scalar_mix" in n and "scalar_mix_0" not in n
-            ]
-            # Fails when sep_embs_for_skip is 0 and elmo_scalars has nonzero
-            # length.
-            assert_for_log(
-                not elmo_scalars or args.sep_embs_for_skip,
-                "Error: ELMo scalars loaded and will be updated in do_target_task_training but "
-                "they should not be updated! Check sep_embs_for_skip flag or make an issue.",
-            )
+        pre_target_train_model_state = setup_target_task_training(args, target_tasks, model, strict)
```

Comment:
Note: We now only load the checkpoint to return back to once instead of every time for each task.

Reply:
Probably too expensive. :/ 

Suggested logic:
- setup returns a path
- when training on the first task (enumerate and test i == 0), load the path
- when training on later tasks, load the path only if paradigm == finetune